### PR TITLE
[Rest Api Compatibility] Enable tests after types and cat api fixed

### PR DESCRIPTION
**Pull request by pgomulka**
Some tests are fixed after typed api is available with compatible api.
Also cat api returning text fixed some tests
relates #51816

### CODE REVIEW

File: x-pack/plugin/watcher/qa/rest/build.gradle [hunk 1]
```diff
@@ -52,13 +52,9 @@ tasks.named("transformV7RestTests").configure({ task ->
 
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
-    'mustache/10_webhook/Test webhook action with mustache integration',
```
Comment (pgomulka) [inline x-pack/plugin/watcher/qa/rest/build.gradle:55]: https://gist.github.com/pgomulka/753da1ea4cb303d313216ced449a81b1
@jakelandis do you have any ideas why is this test failing? Does the stacktrace or logs (see gist) give you any clues?

File: modules/ingest-user-agent/build.gradle [hunk 2]
```diff
@@ -25,7 +25,10 @@ testClusters.all {
...
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
-    'ingest-useragent/30_custom_regex/Test user agent processor with custom regex file',
-    'ingest-useragent/20_useragent_processor/Test user agent processor with parameters'
+//    'ingest-useragent/30_custom_regex/Test user agent processor with custom regex file',
+//    'ingest-useragent/20_useragent_processor/Test user agent processor with parameters'
   ].join(',')
 }
+tasks.named("transformV7RestTests").configure({ task ->
+  task.addAllowedWarningRegex("setting \\[ecs\\] is deprecated as ECS format is the default and only option")
```
Comment (pgomulka) [inline modules/ingest-user-agent/build.gradle:33]: a deprecation added in 8.0

File: modules/analysis-common/build.gradle [hunk 3]
```diff
@@ -25,11 +25,18 @@ dependencies {
...
+}
+
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
-    'search.query/50_queries_with_synonyms/Test common terms query with stacked tokens',
-    'indices.analyze/10_analyze/htmlStrip_deprecated',
-    'analysis-common/40_token_filters/delimited_payload_filter_error',
-    'analysis-common/20_analyzers/standard_html_strip'
-  ].join(',')
+    'search.query/50_queries_with_synonyms/Test common terms query with stacked tokens', // #42654 - not sure, remove `common` query
```
Comment (pgomulka) [inline modules/analysis-common/build.gradle:39]: there is more tests that could be fixed by having https://github.com/elastic/elasticsearch/pull/42654 available under compatible api.
@jimczi this item is marked as `not_sure`. Do you have any views on implementing compatible api for it?

File: rest-api-spec/build.gradle [hunk 4]
```diff
@@ -95,21 +97,17 @@ tasks.named("yamlRestCompatTest").configure {
...
     'mlt/30_unlike/Basic mlt query with unlike',
-    'search.aggregation/200_top_hits_metric/top_hits aggregation with sequence numbers',
-    'search.aggregation/20_terms/*profiler*', // The profiler results aren't backwards compatible.
-    'search.aggregation/51_filter_with_types/Filter aggs with terms lookup and ensure it\'s cached',
-    'search.aggregation/370_doc_count_field/Test filters agg with doc_count', // Uses profiler for assertions which is not backwards compatible
+    'search.aggregation/200_top_hits_metric/top_hits aggregation with sequence numbers', //Marked as not-sure Remove deprecated sort options: nested_path and nested_filter #42809
     'mtermvectors/11_basic_with_types/Basic tests for multi termvector get',
     'mtermvectors/21_deprecated_with_types/Deprecated camel case and _ parameters should fail in Term Vectors query',
     'mtermvectors/30_mix_typeless_typeful/mtermvectors without types on an index that has types',
-    'search/150_rewrite_on_coordinator/Ensure that we fetch the document only once', //terms_lookup
```
Comment (pgomulka) [inline rest-api-spec/build.gradle:105]: already fixed, added back here by mistake due to merge conflict

File: x-pack/plugin/watcher/qa/rest/build.gradle (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/watcher/qa/rest/build.gradle:55]: Another variant of the the types removal. In this case Watcher is writing to ES via a webhook action and using types. Types are incidental to this test, so we could have either removed it in 7.x, thereby fixing it here. However, this is a good use case to implement an arbitrary key/value find/replace that leverages the existing `ReplaceTextual` . I put up a PR over at https://github.com/elastic/elasticsearch/pull/75193 for the fix.

**Review by jakelandis — APPROVED** [review]
LGTM. Once all of the tests we expect to pass we should remove the "v7compatibilityNotSupportedTests" array since it will be redundant.

**Pull request merged**